base/compiler/abstractinterpretation.jl

```diff
@@ -547,7 +547,13 @@ function collect_slot_refinements(𝕃ᵢ::AbstractLattice, applicable::Vector{A
             sigt = Bottom
             for j = 1:length(applicable)
                 match = applicable[j]::MethodMatch
-                sigt = sigt ⊔ fieldtype(match.spec_types, i)
+                spect = fieldtype(match.spec_types, i)
+                if isType(spect)
```

Comment:
(this would be `spect isa Type`, though `valid_as_lattice(spect, true)` is just a more advanced version of that)